### PR TITLE
Rename master to main

### PR DESCRIPTION
**Pull request by ikarldasan**
**What this PR does / why we need it**:
Needed for https://github.com/kubernetes/k8s.io/issues/1597
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Needed for https://github.com/kubernetes/k8s.io/issues/1597

### CODE REVIEW

**Comment by sbueringer** [comment]
@ikarldasan Thx for the PR, unfortunately the changes in this PR are not all that are needed to change the default branch in CAPO. I will implement the other changes directly after the v0.4.0 release and then also merge you're PR here.

/ok-to-test

File: RELEASE.md [hunk 1]
```diff
@@ -48,12 +48,12 @@
 
 Releasing requires a particular set of permissions.
 
-* Approver role for the image promoter process ([kubernetes/k8s.io/blob/master/k8s.gcr.io/images/k8s-staging-capi-openstack/OWNERS](https://github.com/kubernetes/k8s.io/blob/master/k8s.gcr.io/images/k8s-staging-capi-openstack/OWNERS))
+* Approver role for the image promoter process ([kubernetes/k8s.io/blob/main/k8s.gcr.io/images/k8s-staging-capi-openstack/OWNERS](https://github.com/kubernetes/k8s.io/blob/master/k8s.gcr.io/images/k8s-staging-capi-openstack/OWNERS))
```
Comment (tobiasgiese) [inline RELEASE.md:51]:
```suggestion
* Approver role for the image promoter process ([kubernetes/k8s.io/blob/main/k8s.gcr.io/images/k8s-staging-capi-openstack/OWNERS](https://github.com/kubernetes/k8s.io/blob/main/k8s.gcr.io/images/k8s-staging-capi-openstack/OWNERS))
```

File: RELEASE.md [hunk 2]
```diff
@@ -48,12 +48,12 @@
...
 * Tag push and release creation rights to the GitHub repository (team `cluster-api-provider-openstack-maintainers` in [kubernetes/org/config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml](https://github.com/kubernetes/org/blob/master/config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml))
 
 ## Staging
 
 There is a post-submit Prow job running after each commit on master which pushes a new image to the staging repo (`gcr.io/k8s-staging-capi-openstack/capi-openstack-controller:latest`). Following configuration is involved:
-* staging gcr bucket: [kubernetes/k8s.io/blob/master/k8s.gcr.io/manifests/k8s-staging-capi-openstack/promoter-manifest.yaml](https://github.com/kubernetes/k8s.io/blob/master/k8s.gcr.io/manifests/k8s-staging-capi-openstack/promoter-manifest.yaml)
+* staging gcr bucket: [kubernetes/k8s.io/blob/main/k8s.gcr.io/manifests/k8s-staging-capi-openstack/promoter-manifest.yaml](https://github.com/kubernetes/k8s.io/blob/master/k8s.gcr.io/manifests/k8s-staging-capi-openstack/promoter-manifest.yaml)
```
Comment (tobiasgiese) [inline RELEASE.md:57]:
```suggestion
* staging gcr bucket: [kubernetes/k8s.io/blob/main/k8s.gcr.io/manifests/k8s-staging-capi-openstack/promoter-manifest.yaml](https://github.com/kubernetes/k8s.io/blob/main/k8s.gcr.io/manifests/k8s-staging-capi-openstack/promoter-manifest.yaml)
```

File: RELEASE.md (hunk 1 above)
Comment (ikarldasan) [reply RELEASE.md]: Addressed the URL

File: RELEASE.md (hunk 2 above)
Comment (ikarldasan) [reply RELEASE.md]: Addressed the URL

**Review by spiffxp — APPROVED** [review]
/ok-to-test
/lgtm

**Pull request merged**